src/Schema/AbstractIntrospectingSchemaManager.php

```diff
@@ -0,0 +1,170 @@
+<?php
+
+namespace Doctrine\DBAL\Schema;
+
+use Doctrine\DBAL\Exception;
+use Doctrine\DBAL\Exception\DatabaseRequired;
+use Doctrine\DBAL\Platforms\AbstractPlatform;
+use Doctrine\DBAL\Result;
+
+use function array_shift;
+
+/**
+ * Base class for schema managers that improves database schema introspection performance.
```

Comment:
Speaking of improvements makes sense in a commit message or PR, but in code, one might wonder what is the slow version of this.

```suggestion
 * Base class for schema managers that provides good database schema introspection performance.
```

Reply:
Maybe, just avoid qualitative statements at all

>  Base class for schema managers that provides database schema introspection.